### PR TITLE
Respect DebugSymbols=false Fixes #41364

### LINKED ISSUE #41364: -p:DebugSymbols=false does not disable generation of .pdb files
### Describe the bug

https://learn.microsoft.com/en-us/visualstudio/msbuild/common-msbuild-project-properties?view=vs-2022  states:

> Setting -p:DebugSymbols=false on the command line disables generation of program database (.pdb) symbol files.

This does not match .NET 8 SDK behavior.  

`-p:DebugType=None` is the actual property that can be used to disable the .pdb symbol generation.

DebugSymbols property should be either documented as obsolete and no longer supported with DebugType as recommended replacement; or the behavior DebugSymbols property should be fixed to match the documentation.

### To Reproduce

```
dotnet new console
dotnet build -p:DebugSymbols=false
dir /s *.pdb
```

Actual result: .pdb file is generated
Expected result: No .pdb file is generated

### Further technical details
.NET SDK 8.0.205

### PR DESCRIPTION
**Pull request by Forgind**
Fixes #41364

DebugSymbols=false is a parameter that ultimately gets passed to the compiler, disabling pdb generation. There's also DebugType, which, when set to None, is supposed to disable generating pdbs. When DebugSymbols is set but not DebugType, the SDK sees DebugType as not set and sets it to portable, leading to pdb generation even though DebugSymbols was set to false. This is unexpected, so this makes DebugType automatically None when DebugSymbols was false.

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props [hunk 1]
```diff
@@ -94,6 +94,7 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
 
   <PropertyGroup>
+    <DebugType Condition="'$(DebugSymbols)' == 'false'">None</DebugType>
```
Comment (Forgind) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props:97]: I was undecided as to what I should do if DebugType already has a value. Open to suggestions.

Comment (nagilson) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props:97]: I think `DebugSymbols` should take precedence. Honestly its very opinionated but the philosophy of the property / the whole point it exists is to disable the pdbs, so if it doesn't work when the other property is set it feels not so useful.

Furthermore, `DebugSymbols` has been around longer than `DebugType`. DebugSymbols is at least 8 years old.
https://github.com/MicrosoftDocs/visualstudio-docs/commit/381d67e

**Review by nagilson — COMMENTED** [review]
The change looks good, but it would be nice to have a test to show this behavior and prevent regression.

This is also technically a breaking change, so we should be careful about that. Added the breaking-change doc label.

**Review by nagilson — APPROVED** [review]

**Comment by Forgind** [comment]
> This is also technically a breaking change

It's a user-visible change in behavior, but I believe this is going back to how things were before .NET 8, so I believe it's a fix for a regression rather than a new breaking change. I didn't actually check that, though; I'm just going off of what the bug says.

**Comment by nagilson** [comment]
That's true, but it's certainly possible someone started relying on this behavior in .NET 8, setup an automated pipeline, and did not notice they had DebugSymbols=false. Then they would be pretty upset when they eventually find out that the pdb is no longer generated. I think this is the right change and it's good to have, _but_ for almost any behavioral change in the SDK, someone is going to get broken by it 🥹

**Comment by Forgind** [comment]
@dotnet-policy-service rerun

**Comment by rainersigwald** [comment]
> I believe this is going back to how things were before .NET 8

What's that based on @Forgind? I don't think the behavior is that new.

**Comment by Forgind** [comment]
> > I believe this is going back to how things were before .NET 8
> 
> What's that based on @Forgind? I don't think the behavior is that new.

jkotas said the documentation didn't match .NET 8 behavior, which suggests that it changed a bit before .NET 8. To be clear, I wasn't saying it changed in .NET 8 or even .NET 7; I'm not really sure when 'before' .NET 8 it changed to this behavior. Because I didn't know, we ultimately decided to classify this as a standard breaking change and retarget to 9.0 accordingly.

**Pull request merged**